### PR TITLE
Allow graph parameters for reshapes target size

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/data_ops.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "tensorflow/compiler/xla/client/xla_builder.h"
 #include "tensorflow/core/lib/gtl/array_slice.h"
 #include "torch/csrc/jit/ir.h"
+#include "torch_xla/csrc/translator.h"
```
Comment (dlibenzi) [inline torch_xla/csrc/data_ops.h]: Translator depends on data_ops, and data_ops depends on translator?

Comment (asuhan) [reply torch_xla/csrc/data_ops.h]: Translator header doesn't depend on data_ops, only its implementation.

Comment (dlibenzi) [reply torch_xla/csrc/data_ops.h]: OK for this, but the core translator data structures should be in a separate file for cleaner dependency graph.

**Comment by dlibenzi** [comment]
You can leave as is in this PR. We can revisit later.

File: torch_xla/csrc/data_ops.h (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/data_ops.h]: Better? We only need the `SizeOpValues` alias, seems excessive to create a new header for it. Maybe we could move `XlaComputationInOut` and `XlaTranslationResult` to a separate header, but I don't think there's a good justification for it besides the availability of `SizeOpValues` alias.

**Pull request merged**